odoo/models.py

```diff
@@ -1746,6 +1747,108 @@ def _read_group_fill_results(self, domain, groupby, remaining_groupbys,
 
         return list(result.values())
 
+    @api.model
+    def _read_group_fill_temporal(self, data, groupby, aggregated_fields,
+                                  interval=dateutil.relativedelta.relativedelta(months=1)):
+        """Helper method for filling date/datetime 'holes' in a result set.
+
+        We are in a use case where data are grouped by a date field (typically
+        months but it could be any other interval) and displayed in a chart.
+
+        Assume we group records by month, and we only have data for August,
+        September and December. By default, plotting the result gives something
+        like:
+                                                ___
+                                      ___      |   |
+                                     |   |     |   |
+                                     |   | ___ |   |
+                                     |   ||   ||   |
+                                     |___||___||___|
+                                      Aug  Sep  Dec
+
+        The problem is that December data follows immediately September data,
+        which is misleading for the user. Adding explicit zeroes for missing data
+        gives something like:
+                                                     ___
+                                 ___                |   |
+                                |   |               |   |
+                                |   | ___           |   |
+                                |   ||   |          |   |
+                                |___||___| ___  ___ |___|
+                                 Aug  Sep  Oct  Nov  Dec
+
+        :param list data: the data containing groups
+        :param list groupby: name of the first group by
+        :param aggregated_fields list: list of aggregated fields in the query
+        :param relativedelta interval: interval between to temporal groups
+        expressed as a relativedelta month by default
+        :rtype: list
+        :return: list
+        """
+        groupby_name = groupby[0]
+
+        # existing non null datetimes
+        existing = [d[groupby_name] for d in data if d[groupby_name]]
+
+        if len(existing) < 2:
+            return data
+
+        # assumption: existing data is sorted by field 'groupby_name'
+        first, last = existing[0], existing[-1]
+
+        # When the date_trunc function is used on a DATE (not a TIMESTAMP),
+        # PostGreSQL implicitly cast it as a TIMESTAMP WITH TIMEZONE. That time-
+        # zone is based on the local machine executing Postgres. We don't want
+        # this because it could cause various problems. That's why we detect
+        # those TZs by assuming they are always an instance of
+        # psycopg2.tz.FixedOffsetTimezone and remove them. Other possible
+        # timezone should come from the user context.
+        remove_tz = isinstance(first.tzinfo, psycopg2.tz.FixedOffsetTimezone)
```

Comment:
replace this hack by the proper casting to `::timestamp` [here](https://github.com/odoo/odoo/blob/e7b973b95326dd2f886d51537669b50fa09a08ba/odoo/models.py#L1942), so you can also remove other code dealing with `remove_tz` below